fs-index/src/tests.rs

```diff
@@ -666,3 +666,245 @@ fn test_update_all_files_with_same_hash() {
         assert_eq!(index.collisions().len(), 1, "{:?}", index);
     });
 }
+
+/// Simple test for tracking a single resource addition.
```

Comment:
In the `arklib` PR we also tried to implement an automated test scenario, where we'd test one specific invariant that "any sequence of `update_one` calls must be equivalent to `update_all` on the full data". My idea how to approach it is something like:
1. Start with empty directory.
2. Generate an update, randomly one of 3 kinds (modification and deletion could be generated only when we have some files). 
3. Call `update_one` on the updated file.
4. In isolation, build new index from the current state of the directory.
5. Compare outputs of step 3 and step 4. Must be equal.

This is not exactly a unit test, because it's slow and tests multiple corner cases in one scenario. But I think it must be useful to run it periodically. By the way, I think that Ishan and Pushkar do in `fs-storage` with property-based testing should reflect the same idea.

Do you think it's feasible to implement in this PR?

Reply:
Sounds like a plan.
We could create multiple tests to verify this as well. 
To do this properly and cleanly, we’ll likely need to set up some helper functions. 

Given the scope, I think this should be handled in its own PR. This current PR is blocking #36, so once it's merged, I can add the additional tests along with the index watch API. 

What do you think?